### ISSUE TITLE
L dummy option

### ISSUE THREAD

**Issue by iwahiropuzzles** [kkondo1981/aglm#34]
By using O dummies, the current version implements "fusion" (fused LASSO) not only for ordered factor variables but also for numeric variables. But, for numeric variables, "linear interpolation" may be a good, possibly better, alternative.

My idea is this:
1. Outline
For numeric variables, add an option of using new dummies, say, L dummies, instead of using O dummies and linear terms, where L dummies are dummies for linear interpolation. (Don't use linear terms and linear interpolation at the same time.)
2. Details
(i) L dummies can be implemented by adding a new function, say, `getLDummyMatForOneVec`.
The code is to be something like the following. (It's same as `getODummyMatForOneVec` but with only differences in the lines with #'s on the right.)

```
getLDummyMatForOneVec <- function(x_vec, breaks=NULL, nbin.max=100, only_info=FALSE) {
 # Check arguments. only integer or numerical or ordered vectors are allowed.
 assert_that(is.integer(x_vec) | is.numeric(x_vec) | is.ordered(x_vec))

 # Execute binning
 binned_x <- executeBinning(x_vec, breaks=breaks, nbin.max=nbin.max, method="width") #

 # create dummy matrix for x_vec
 nrow <- length(x_vec)
 ncol <- length(binned_x$breaks)
 dummy_mat <- (binned_x$labels - t(matrix(1:ncol, ncol, nrow))) * (binned_x$labels > t(matrix(1:ncol, ncol, nrow))) #

 if (only_info) return(list(breaks=binned_x$breaks))
 else return(list(breaks=binned_x$breaks, dummy_mat=dummy_mat))
} 
```
  (ii) For plot-aglm.R, the corresponding part for L dummies is to be something like this (see. https://github.com/kkondo1981/aglm/blob/c6570d72e011107bb030e26daada09cc0782229f/R/plot-aglm.R#L23-L31)
```
     # Plot for numeric features
      slopesteps <- coefs$coef.LD
      if (is.null(slopesteps)) slopesteps <- 0

      x <- var_info$LD_info$breaks
      y <- cumsum(cumsum(slopesteps))
      type <- "l"
```

**Comment by kkondo1981** [comment]
As a first step, I implemented continuous type of O-dummies in #35 , whose contribution curve (say, f(x) = OD(x) \cdot \beta ) are continuous and have no jumps contrary to the original O-dummies.

Because more smooth contribution curves using L-dummies may be useful, I will keep this issue open, but think it's future work.

**Comment by kkondo1981** [comment]
Merged @kazuzowo 's LVar features into #49
